INPTP_AppForFixing/MainForm.cs

```diff
@@ -44,18 +53,37 @@ public void OnEmployeeChange()
             if (CheckSelectedBoss())
             {
                 Boss temp = listBoxOfBosses.SelectedItem as Boss;
-                if (temp.GetEmployees().Count > 0)
+                if (temp.GetEmployees().Count >= 0)
                 {
                     listBoxEmpl.DataSource = new List<Employee>(temp.GetEmployees());
-                }                
-            }            
+                }
+            }
         }
 
+        public Boss GetSelectedBoss() => (Boss)listBoxOfBosses.SelectedItem;
+
+        public Employee GetSelectedEmployee() => (Employee)listBoxEmpl.SelectedItem;
+
+
+        public int getNextEmployeeId()
+        {
+            try
+            {
+                return Math.Max(bosses.Max(b => b.Id), bosses.Max(b => b.GetEmployees().Any() ? b.GetEmployees().Max(e => e.Id) : int.MinValue)) + 1;
```

Comment:
Unhandled case - employee in boss in boss... And similair deeper nested cases

Reply:
What exactly do you mean by that?

Reply:
You could have following structure:
Boss(1)
* Employee(2)
* Employee(3)
* Employee(4)
* Boss(5)
* * Employee(6)
* * Employee(7) 
* * Boss(8)
* * * Employee(9)
* Employee(10)

Current check wont check ids 6, 7, 8, 9.

Reply:
fixed